### PR TITLE
Re-Grow - limb regeneration for campaign

### LINKED ISSUE #16817: Som engineer gets TGMC setrnty ammo
### Description

if you pick the cope sentry in the loadouts for campain you'll get the sentry for it but also the ammo for the ST-580 sentry

### Test Merges

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/sybil/server_side_modifications.dm

Round ID: [33225](https://scrubby.melonmesa.com/round/33225)
BYOND Version: 515.1633
Server revision compiled on: 2024-12-14T20:46:12+00:00
Master commit: [baf91554e327ca8ae05cdb621574dda2120276f9](https://github.com/tgstation/TerraGov-Marine-Corps//commit/baf91554e327ca8ae05cdb621574dda2120276f9)
The following pull requests are currently test merged:
[#16674: 'squad/Radio TTS fix' by Lumipharon at commit 240cead51c](https://github.com/tgstation/TerraGov-Marine-Corps//pull/16674)
[#16662: 'TTS order announcements/hivemind' by TiviPlus at commit 6d5402d1e8](https://github.com/tgstation/TerraGov-Marine-Corps//pull/16662)
[#16504: 'Metnal Mining Operation (REVIVAL)' by PibeCantina at commit d898d1805b](https://github.com/tgstation/TerraGov-Marine-Corps//pull/16504)
[#16725: 'Adds new marine race: Prototype Supersoldier' by TiviPlus at commit 25f3edbbde](https://github.com/tgstation/TerraGov-Marine-Corps//pull/16725)
[#16664: 'Adds bloodthirster Rav strain' by TiviPlus at commit f0f46930df](https://github.com/tgstation/TerraGov-Marine-Corps//pull/16664)

Local commit: 600bf57711612f508be65df301ae4859934a53d9
Server tools version: 6.9.0

### Reproduction Steps

play SOM engineer pick sentry

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Added Re-Grow injectors to medic's advanced chem secondary kit.
The injector has 6 doses, each dose on inject will regenerate ALL missing limbs on the subject. New limbs are biotic however, so are weaker than standard.

Also fixed a couple of minor loadout bugs.
Fixed duplicate 'extra ammo' secondary for medics, and it not being the default choice as intended.
Fixes #16817 The COPE sentry secondary having TGMC sentry ammo in it.

## Why It's Good For The Game
I still prefer for delimbs to be in HvH, but some weapons make it a bit too easy to achieve. This adds some counterplay and further increases the usefulness of medics.

## Changelog

:cl:
add: Campaign: Added Re-Grow to advanced med secondary kit, allowing for regeneration of missing limbs
fix: Campaign: Fixed a couple of minor secondary kit bugs
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medical.dm [hunk 1]
```diff
@@ -1336,3 +1336,44 @@
...
+/datum/reagent/medicine/regrow
+	name = "Re-grow"
+	description = "Re-grow is rare and unusual drug that stimulates the rapid (and horrifically painful) regeneration of missing limbs."
+	color = COLOR_REAGENT_SYNAPTIZINE
+	overdose_threshold = REAGENTS_OVERDOSE/5
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5
+	custom_metabolism = REAGENTS_METABOLISM * 5
+
+/datum/reagent/medicine/regrow/on_mob_add(mob/living/L, metabolism)
+	if(volume > 5 || L.stat == DEAD || (!ishuman(L)))
```
Comment (QualityVan) [inline code/modules/reagents/chemistry/reagents/medical.dm:1349]: Probably should be < 5?

File: code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm [hunk 2]
```diff
@@ -139,7 +139,7 @@
 
 /datum/loadout_item/secondary/kit/som_corpsman/advanced
 	name = "Advanced meds"
-	desc = "A variety of advanced medical injectors including Russian Red, as well as rezadone, a powerful chemical able to treat genetic damage in humanoids."
+	desc = "A variety of advanced medical injectors including Russian Red, rezadone and Re-Grow, allowing for the treatment cloneloss and missing limbs."
```
Comment (Arturlang) [inline code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm:142]: treatment of cloneloss

File: code/modules/reagents/chemistry/reagents/medical.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/reagents/chemistry/reagents/medical.dm]: fixed

File: code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm (hunk 2 above)
Comment (Lumipharon) [reply code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm:142]: ye?

Comment (QualityVan) [reply code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm:142]: as in the word 'of' is missing right now

**Comment by Technical-l** [comment]
science wins again

File: code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm (hunk 2 above)
Comment (Lumipharon) [reply code/datums/gamemodes/campaign/loadout_items/SOM/secondaries.dm:142]: oh @Arturlang just make a change suggestion nerd, I can't read.

**Review by QualityVan — APPROVED** [review]

**Pull request merged**